client/components/domains/register-domain-step/index.jsx

```diff
@@ -646,24 +704,24 @@ class RegisterDomainStep extends React.Component {
 	};
 
 	onSearch = searchQuery => {
+		debug( 'onSearch handler was triggered with query', searchQuery );
 		const domain = getFixedDomainSearch( searchQuery );
 
 		this.setState(
 			{ lastQuery: searchQuery, lastSurveyVertical: this.props.surveyVertical },
 			this.save
 		);
 
-		if ( ! domain || ! this.state.loadingResults ) {
+		if ( ! domain || ( typeof domain === 'string' && domain.trim() === '' ) ) {
```

Comment:
When is this case possible? Seems `getFixedDomainSearch` above takes care of trimming and such (and in general ensuring it's a proper string).

Reply:
Having looked at `getFixedDomainSearch`, I don't think it's possible. I'll simplify this to `domain === ''`.